src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,223 @@
+    public Car create(Car car) {
+        String insertFormatQuery =
+                "INSERT INTO cars (model, manufacturer_id) VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createStatement =
+                        connection.prepareStatement(insertFormatQuery,
+                             Statement.RETURN_GENERATED_KEYS)) {
+            createStatement.setString(1, car.getModel());
+            createStatement.setLong(2, car.getManufacturer().getId());
+            createStatement.executeUpdate();
+            ResultSet resultSet = createStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create "
+                    + car + ". ", e);
+        }
+        insertDriver(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getQuery =
+                "SELECT c.id AS car_id, mf.name, model,country, manufacturer_id "
+                        + "FROM cars c JOIN manufacturers mf "
+                        + "ON c.manufacturer_id = mf.id "
+                        + "WHERE c.id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getByIdStatement =
+                        connection.prepareStatement(getQuery)) {
+            getByIdStatement.setLong(1, id);
+            ResultSet resultSet = getByIdStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAll =
+                "SELECT c.id AS car_id, model, m.id AS manufacturer_id, m.name, country "
+                        + "FROM cars c "
+                        + "JOIN manufacturers m "
+                        + "ON c.manufacturer_id = m.id "
+                        + "AND c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getAll)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from DB.", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversForCar(car.getId()));
```

Comment:
The same. I think you can move `car.setDrivers(getDriversForCar(id))` inside the method `getCar(resultSet)`, and get rid from a redundant second loop.

Reply:
and here I also was wrong ))